### ISSUE TITLE
The local cache artifact in "*" was not been generated on this machine. >5.0.0

### ISSUE THREAD

**Issue by Squixx** [NiklasPor/nx-remotecache-custom#26]
>  NX   Invalid Cache Directory for Task "*-e2e:e2e"

   The local cache artifact in "*" was not been generated on this machine.
   As a result, the cache's content integrity cannot be confirmed, which may make cache restoration potentially unsafe.
   If your machine ID has changed since the artifact was cached, run "nx reset" to fix this issue.
   Read about the error and how to address it here: https://nx.dev/recipes/troubleshooting/unknown-local-cache
   
   Pass --verbose to see the stacktrace.

Reproduction steps:

Dont have a local cache, but a remote available. Run the task so it retrieves from cache. Run the task again and you get invalid cache dir.
```
    "nx": "16.10.0",
    "@nx/workspace": "16.10.0",

    ```

```
 at Cache.assertLocalCacheValidity (*/node_modules/.pnpm/nx@16.10.0_@swc-node+register@1.6.8_@swc+core@1.3.92/node_modules/nx/src/tasks-runner/cache.js:203:23)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Cache.get (*/node_modules/.pnpm/nx@16.10.0_@swc-node+register@1.6.8_@swc+core@1.3.92/node_modules/nx/src/tasks-runner/cache.js:58:13)
    at async TaskOrchestrator.applyCachedResult (*s/node_modules/.pnpm/nx@16.10.0_@swc-node+register@1.6.8_@swc+core@1.3.92/node_modules/nx/src/tasks-runner/task-orchestrator.js:79:30)
    at async Promise.all (index 0)
    at async TaskOrchestrator.applyCachedResults */node_modules/.pnpm/nx@16.10.0_@swc-node+register@1.6.8_@swc+core@1.3.92/node_modules/nx/src/tasks-runner/task-orchestrator.js:74:21)
    at async TaskOrchestrator.applyFromCacheOrRunTask (*/node_modules/.pnpm/nx@16.10.0_@swc-node+register@1.6.8_@swc+core@1.3.92/node_modules/nx/src/tasks-runner/task-orchestrator.js:153:40)
    at async TaskOrchestrator.executeNextBatchOfTasksUsingTaskSchedule (*/node_modules/.pnpm/nx@16.10.0_@swc-node+register@1.6.8_@swc+core@1.3.92/node_modules/nx/src/tasks-runner/task-orchestrator.js:64:13)
    at async Promise.all (index 0)
    at async TaskOrchestrator.run (*/node_modules/.pnpm/nx@16.10.0_@swc-node+register@1.6.8_@swc+core@1.3.92/node_modules/nx/src/tasks-runner/task-orchestrator.js:41:9)
```

note: the cache needs to be created on another machine

**Comment by NiklasPor** [comment]
@Squixx this is fixed in all packages at `5.0.1` 👍 Thanks for reporting.

**Comment by Squixx** [comment]
Thats fast.. awesome.
